system-int-test/test/cmr/system_int_test/search/service/collection_service_search_test.clj

```diff
@@ -0,0 +1,131 @@
+(ns cmr.system-int-test.search.service.collection-service-search-test
+  "Tests searching for collections with associated services"
+  (:require
+   [clojure.test :refer :all]
+   [cmr.common.util :refer [are3]]
+   [cmr.mock-echo.client.echo-util :as e]
+   [cmr.system-int-test.data2.atom :as atom]
+   [cmr.system-int-test.data2.collection :as dc]
+   [cmr.system-int-test.data2.core :as d]
+   [cmr.system-int-test.system :as s]
+   [cmr.system-int-test.utils.association-util :as au]
+   [cmr.system-int-test.utils.index-util :as index]
+   [cmr.system-int-test.utils.ingest-util :as ingest]
+   [cmr.system-int-test.utils.search-util :as search]
+   [cmr.system-int-test.utils.service-util :as service-util]))
+
+(use-fixtures :each
+              (join-fixtures
+               [(ingest/reset-fixture {"provguid1" "PROV1"})
+                service-util/grant-all-service-fixture]))
+
+(defn- verify-collection-atom-has-formats
+  "Verify the collection ATOM response has-formats field has the given value"
+  [coll has-formats]
+  (let [coll-with-extra-fields (assoc coll :has-formats has-formats)
+        {:keys [entry-title]} coll
+        coll-atom (atom/collections->expected-atom
+                   [coll-with-extra-fields]
+                   (format "collections.atom?entry_title=%s" entry-title))
```

Comment:
I find this confusing to pass in the URL. I don't think in most cases we care about the :id being returned from an ATOM search (we should verify it in some test, but we shouldn't have to in every test). But then I guess you'd have to perform a dissoc of the id when performing the comparison.

Reply:
I didn't care too much. This way is just easy to copy and paste for me and it is consistent. I guess consistency is good. :)